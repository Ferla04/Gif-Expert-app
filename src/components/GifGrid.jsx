import React from 'react'

export const GifGrid = ({ category }) => {


  getGifs(category)


  return (
    <>
      <h3>{ category }</h3>
    </>
  )
}


const getGifs = async () => {

  const url = `https://api.giphy.com/v1/gifs/search?api_key=k0WUNnR1P5buYYsugmhELhe7uPLW7L17&q=${category}&limit=20`
  const resp = await fetch(url)
  const { data = [] } = await resp.json()

  const gifs = data.map( img => ({
    id: img.id,
    title: img.title,
    url: img.images.downsized_medium.url
  }))

  return gifs
}