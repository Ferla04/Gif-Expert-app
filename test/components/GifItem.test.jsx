import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en <GifItem />', () => {

  const title = 'Attack on titan'
  const url = 'https://attack-on-titan/levi.jpg'

  test('Debe hacer match con el snapshot', () => {

    const { container } = render( <GifItem title={ title } url={ url } /> )
    expect( container ).toMatchSnapshot()
  })

  test('Debe mostrar la imagen con el URL y el ATL indicado', () => {

    render( <GifItem title={ title } url={ url } /> )
    // screen.debug()
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( title )
  })

  test('Debe mostar el titulo en el componente', () => {

    render( <GifItem title={ title } url={ url } /> )
    expect( screen.getByText( title ) ).toBeTruthy()

  })
})