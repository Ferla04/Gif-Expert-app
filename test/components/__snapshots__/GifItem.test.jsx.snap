// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Attack on titan"
      src="https://attack-on-titan/levi.jpg"
    />
    <p>
      Attack on titan
    </p>
  </div>
</div>
`;
